src/js/tools/metroconfig.ts

```diff
@@ -158,26 +158,83 @@ function withSentryDebugId(config: MetroConfig): MetroConfig {
   };
 }
 
+type ResolverThreeParams = (
+  context: CustomResolutionContext,
+  moduleName: string,
+  platform: string | null,
+) => Resolution;
+type ResolverFourParams = (
+  context: CustomResolutionContext,
+  moduleName: string,
+  platform: string | null,
+  realModuleName?: string,
+) => Resolution;
+
 /**
  * Includes `@sentry/replay` packages based on the `includeWebReplay` flag and current bundle `platform`.
  */
 export function withSentryResolver(config: MetroConfig, includeWebReplay: boolean | undefined): MetroConfig {
-  const originalResolver = config.resolver?.resolveRequest;
+  const originalResolver = config.resolver?.resolveRequest as ResolverThreeParams | ResolverFourParams;
+
+  const hasSentryReplay = (platform: string | null, moduleName: string): boolean => {
+    return (
+      (includeWebReplay === false || (includeWebReplay === undefined && platform !== 'web')) &&
+      moduleName.includes('@sentry/replay')
+    );
+  };
+
+  let resolver: ResolverThreeParams | ResolverFourParams;
+
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const metro = require('metro/package.json') as { version: string };
+  const [major, minor] = metro.version.split('.').map(Number);
+
+  if (minor >= 68 || major >= 1) {
+    // New method introduced on metro 0.68 and newer.
+    resolver = (context: CustomResolutionContext, moduleName: string, platform: string | null) => {
+      if (hasSentryReplay(platform, moduleName)) {
+        return { type: 'empty' } as Resolution;
+      }
+      if (originalResolver) {
+        return originalResolver(context, moduleName, platform);
+      }
+      return context.resolveRequest(context, moduleName, platform);
+    };
+  } else {
+    // On older Metro, the given context from resolver is not the defaultResolver but the called function itself.
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-var-requires, import/no-extraneous-dependencies
+    const defaultResolver = require('metro-resolver').resolve;
```

Comment:
Let's only require this when `originalResolver` is `undefined`

In case there is an issue with the require so users have a way out.

Let's also do the require in `try catch` and on error let's log out from what path we tried the resolver and where can user open issue. And how to opt out -> setting `includeWebReplay=true`

similar to https://github.com/getsentry/sentry-react-native/blob/13a38606494c437f28c974fc2544a583d5a8f4f1/packages/core/src/js/tools/vendor/metro/utils.ts#L99

and on success let's once (the first time) log out that we are using the fallback and the path.

Reply:
atm users can define the resolveRequest on MetroConfig to not enable it or even set includeWebReplay as true so that withSentryResolver is not executed.